implementations/micrometer-registry-statsd/src/test/java/io/micrometer/statsd/StatsdMeterRegistryPublishTest.java

```diff
@@ -0,0 +1,234 @@
+        counter.increment();
+        counter.increment();
+        assertThat(latch.await(1, TimeUnit.SECONDS)).isTrue();
+        meterRegistry.close();
+    }
+
+    @ParameterizedTest
+    @EnumSource(StatsdProtocol.class)
+    void whenBackendInitiallyDown_metricsSentAfterBackendStarts(StatsdProtocol protocol) throws InterruptedException {
+        serverLatch = new CountDownLatch(3);
+        // start server to secure an open port
+        DisposableChannel server = startServer(protocol, 0);
+        final int port = server.address().getPort();
+        server.disposeNow();
+        meterRegistry = new StatsdMeterRegistry(getUnbufferedConfig(protocol, port), Clock.SYSTEM);
+        meterRegistry.start();
+        Counter counter = Counter.builder("my.counter").register(meterRegistry);
+        IntStream.range(0, 100).forEach(counter::increment);
+        server = startServer(protocol, port);
+        await().until(() -> meterRegistry.client.get() != null);
+        // TcpClient may take some time to reconnect to the server
+        await().until(() -> !clientIsDisposed());
+        assertThat(serverLatch.getCount()).isEqualTo(3);
+        counter.increment();
+        counter.increment();
+        counter.increment();
+        assertThat(serverLatch.await(1, TimeUnit.SECONDS)).isTrue();
+        meterRegistry.close();
+        server.disposeNow();
+    }
+
+    private void startRegistryAndWaitForClient() {
+        meterRegistry.start();
+        // TODO alternatively, configure the processor to buffer when no subscriber is present... previous behavior used to be this, I think?
+        await().until(() -> !clientIsDisposed());
+    }
+
+    private boolean clientIsDisposed() {
+        return meterRegistry.client.get().isDisposed();
+    }
+
+    @NotNull
+    private DisposableChannel startServer(StatsdProtocol protocol, int port) {
+        if (protocol == StatsdProtocol.UDP) {
+            return UdpServer.create()
+                    .host("localhost")
+                    .port(port)
+                    .handle((in, out) ->
+                            in.receive().asString()
+                                    .flatMap(packet -> {
+                                        serverLatch.countDown();
+                                        return Flux.never();
+                                    }))
+                    .wiretap("udpserver", LogLevel.INFO)
+                    .bindNow(Duration.ofSeconds(2));
+        } else if (protocol == StatsdProtocol.TCP) {
+            return TcpServer.create()
+                    .host("localhost")
+                    .port(port)
+                    .handle((in, out) ->
+                            in.receive().asString()
+                                    .flatMap(packet -> {
+                                        IntStream.range(0, packet.split("my.counter").length - 1).forEach(i -> serverLatch.countDown());
+                                        return Flux.never();
```

Comment:
`server.disposeNow()` will not close the active connections. You have to close them either on the client or on the server. On possible solution is the following:

```
.handle((in, out) ->
    in.receive()
       .asString()
       .flatMap(packet -> {
           IntStream.range(0, packet.split("my.counter").length - 1)
                           .forEach(i -> serverLatch.countDown());
           in.withConnection(DisposableChannel::dispose);
           return Flux.never();
           })
)
```

Reply:
This in addition with waiting until the client reconnects before making counter increments cause the TCP tests to pass consistently, in my local testing. Thanks!